lib/matplotlib/backend_managers.py

```diff
@@ -74,6 +73,38 @@ def __init__(self, canvas):
         self.keypresslock = widgets.LockDraw()
         self.messagelock = widgets.LockDraw()
 
+    @property
+    def manager(self):
+        """FigureManager that owns this ToolManager"""
+        return self._manager
+
+    @property
+    def canvas(self):
+        """Canvas managed by FigureManager"""
+        return self._figure.canvas
+
+    @property
+    def figure(self):
+        """Figure that holds the canvas"""
+        return self._figure
+
+    def set_figure(self, figure):
```

Comment:
if I put it as a setter, then when you want to override it you will have to redeclare the property and the setter with it's decorators, if not, the behavior is not the same between the base and the subclass (I can see the problems falling from the sky).

The other option  (the one that I normally use) is the base to the declare the property and its setter with **private** methods that actually do the set and get. And if you need to override this methods, you override the `_set_x` or `_get_x` and the behavior is the same because you don't touch the property

```
@property
def x(self):
   return self._get_x()

@x.setter(self, value):
     self._set_x(value)

def _get_x(self):
    ....

def _set_x(self, value):
    ....
```